plugins/UserCountry/Commands/AttributeHistoricalDataWithLocations.php

```diff
@@ -0,0 +1,341 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Plugins\UserCountry\Commands;
+
+use Piwik\DataAccess\RawLogUpdater;
+use Piwik\Network\IPUtils;
+use Piwik\Plugin\ConsoleCommand;
+use Piwik\Plugins\UserCountry\VisitorGeolocator;
+use Piwik\Plugins\UserCountry\LocationProvider;
+use Piwik\DataAccess\RawLogFetcher;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class AttributeHistoricalDataWithLocations extends ConsoleCommand
```

Comment:
I would actually extract most of the things into a different class(es) and make it testable. There is a lot of logic in here and many possible bugs. A command is pretty much like a controller and should not really contain that much code and not really logic. I will try to make some comments.